posthog/src/main/java/com/posthog/java/QueueManager.java

```diff
@@ -39,35 +40,77 @@ public synchronized void add(JSONObject json) {
         }
     }
 
-    private Sender sending;
     private QueuePtr queue = new QueuePtr();
     private volatile boolean stop = false;
-    // TODO: these should be customizable & set good defaults
-    private int maxQueueSize = 10; // if more than this many items in queue trigger a send
-    private int maxTimeToSendMs = 1000; // if more than this time since last send trigger a send
-    private int sleepMs = 100;
     private Instant sendAfter;
+    // builder inputs
+    private Sender sender;
+    private int maxQueueSize;
+    private Duration maxTimeInQueue;
+    private int sleepMs;
```

Comment:
what's the reason for exposing this?

Reply:
> private Sender sender;

for extensibility (we might provide different Sender's in the future) & testability too. 

>  private int maxQueueSize;
    private Duration maxTimeInQueue;
    private int sleepMs;

So one can configure this to suit their app needs. E.g. if you make `sleepMs` longer it takes less of your cpu with the downside of more delayed sending of data (potential for dropping more of it too). Depending on your network setup you might want to send fewer larger packets over more frequent smaller ones (maxQueueSize & maxTimeInQueue)